contracts/staking/contracts/src/libs/LibProxy.sol

```diff
@@ -0,0 +1,109 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+import "@0x/contracts-utils/contracts/src/LibRichErrors.sol";
+import "./LibStakingRichErrors.sol";
+
+
+library LibProxy {
+
+    enum RevertRule {
+        REVERT_ON_ERROR,
+        ALWAYS_REVERT,
+        NEVER_REVERT
+    }
+
+    /// @dev Executes a read-only call to the staking contract, via `revertDelegateCall`.
+    ///      By routing through `revertDelegateCall` any state changes are reverted.
+    // solhint-disable no-complex-fallback
+    function proxyCall(
+        address destination,
+        RevertRule revertRule,
+        bytes4 customSelector,
+        bool ignoreSelector
+    )
+        internal
+    {
+        if (destination == address(0)) {
```

Comment:
I kind of feel like there's no point checking for this. A wrong address is just as bad as `0x0`.

Reply:
That's true... I guess this is more of an initialization sanity check than a correctness check.

Reply:
We just had a cheeky chat about this in the coder cave. I think we're going to leave these sanity checks in for now (they're used throughout staking) but remove them if we do end up using constants.